lib/content-services/src/lib/legal-hold/services/legal-hold.service.ts

```diff
@@ -0,0 +1,108 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AlfrescoApiService } from '@alfresco/adf-core';
+import { ContentPagingQuery, Hold, HoldEntry, HoldPaging, LegalHoldApi } from '@alfresco/js-api';
+import { Injectable } from '@angular/core';
+import { Observable, from } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class LegalHoldService {
+    private _legalHoldApi: LegalHoldApi;
+    get legalHoldApi(): LegalHoldApi {
+        this._legalHoldApi = this._legalHoldApi ?? new LegalHoldApi(this.apiService.getInstance());
+        return this._legalHoldApi;
+    }
+
+    constructor(private readonly apiService: AlfrescoApiService) {}
+
+    /**
+     * Gets the list of holds available in the file plan.
+     *
+     * @param filePlanId The identifier of a file plan. You can also use the -filePlan- alias.
+     * @param options Optional parameters supported by JS-API
+     * @returns List of holds Observable<Hold[]>
+     */
+    getHolds(filePlanId: string, options?: ContentPagingQuery): Observable<Hold[]> {
+        return from(this.legalHoldApi.getHolds(filePlanId, options)).pipe(
+            map(({ list }) =>
+                list.entries?.map(({ entry }) => ({
+                    id: entry.id,
+                    name: entry.name
+                }))
+            )
+        );
+    }
+
+    /**
+     * Assign a node to a hold.
+     *
+     * @param nodeId The Id of the node which will be assigned to a hold
+     * @param holdId The Id of the hold to which nodes will be assigned
+     * @returns Observable<HoldEntry>
+     */
+    assignHold(nodeId: string, holdId: string): Observable<HoldEntry> {
+        return from(this.legalHoldApi.assignHold(nodeId, holdId));
+    }
+
+    /**
+     * Assign a node to a hold.
+     *
+     * @param nodeIds The list of managed node Ids
+     * @param holdId The Id of the hold to which nodes will be assigned
+     * @returns Observable<HoldPaging>
+     */
+    assignHolds(nodeIds: { id: string }[], holdId: string): Observable<HoldPaging> {
+        return from(this.legalHoldApi.assignHolds(nodeIds, holdId));
+    }
+
+    /**
+     * Unassign the relationship between a child with id nodeId and a parent hold with id holdId.
+     *
+     * @param holdId The hold Id
+     * @param nodeId The Id of the node which is unassigned
+     * @returns Empty response
+     */
+    unassignHold(holdId: string, nodeId: string): Observable<void> {
+        return from(this.legalHoldApi.unassignHold(holdId, nodeId));
+    }
+
+    /**
+     * Create hold.
+     *
+     * @param filePlanId The identifier of a file plan. You can also use the -filePlan- alias.
+     * @param hold Hold to create
+     * @returns List of created holds Observable<HoldEntry>
+     */
+    createHold(filePlanId: string, hold: Hold): Observable<HoldEntry> {
```

Comment:
This implementation is a bit error prone since someone using just this service in ADF may try to create a hold giving it a custom id or other property that might not be supported on the BE, have you tested that case? Does API returns error? If so we should restrict that with other type that will allow to add properties supported during hold creation only